tests/char_healing_test.cpp

```diff
@@ -386,3 +386,106 @@ TEST_CASE( "healing_rate_medicine with bandages and/or disinfectant", "[heal][ba
     }
 }
 
+// Get the character addicted
+void get_addicted( Character &dummy, add_type addiction )
+{
+    // add_addiction uses rng; try several times until addiction is obtained
+    int tries = 0;
+    while( tries < 10 && !dummy.has_addiction( addiction ) ) {
+        dummy.add_addiction( addiction, 20 );
+        ++tries;
+    }
+    REQUIRE( dummy.has_addiction( addiction ) );
+}
+
+// Mending broken bones uses Character::mend_healing_factor to influence the chance of success,
+// with penalties reducing the number below 1.0, and bonuses increasing the number above 1.0.
+//
+// Tobacco and alcohol (and addiction to them) reduce the chance of mending:
+// - Alcohol penalties stack with tobacco penalties.
+// - Being under the influence of alcohol slows mending by 50%.
+// - Being under the influence of tobacco slows mending by 50%.
+// - When not directly under their influence, having an addiction to alcohol and/or tobacco
+//   gives a 25% to 75% penalty to each, depending on severity of addiction.
```

Comment:
75% would mean that someone who's trying to quit could be worse at bone healing than someone who was drunk or smoking 2 packs a day. Huh? I suppose this might make sense for severe alcohol withdrawal (DTs), but while very difficult nicotine withdrawal is not 50% fatal without medical assistance.

Reply:
These are a description of current behavior, to the extent that I understand it. The existing comments in `suffer.cpp` say:

```
// Being addicted but not under effect scales from 25% slowdown to 75% slowdown
// The improvement from being intoxicated over withdrawal is intended
```

I don't think it is fatal; it simply reduces (slows down) the chance of mending. Maybe it is too severe a penalty for a withdrawal symptom, but in this PR I am only interested in testing it.

Reply:
I can see "improvement from being intoxicated over withdrawal" for a lot of things, but not bone healing, particularly for smoking. But I understand about the scope of this PR.